RecoMuon/GlobalTrackingTools/src/MuonTrackingRegionBuilder.cc

```diff
@@ -1,281 +1,278 @@
+     phi = thePhiRegionPar2 + (acoeff_Phi)*(pt-100.);
+  }
+
+  double region_dPhi1 = std::min(phi,dphi);
+  double region_dEta1 = std::min(eta,deta);
+
+  // decide to use either a parametrization or a dynamical region
+  region_dPhi = useFixedPhi ? theDeltaPhi : std::max(thePhiMin,region_dPhi1);
+  region_dEta = useFixedEta ? theDeltaEta : std::max(theEtaMin,region_dEta1);
+
+  float deltaR = theDeltaR;
+  double minPt = useFixedPt ? thePtMin : std::max(thePtMin,pt*0.6);
+
+
+  const MeasurementTrackerEvent* measurementTracker = nullptr;
+  if (!theMeasurementTrackerToken.isUninitialized()) {
+    edm::Handle<MeasurementTrackerEvent> hmte;
+    ev.getByToken(theMeasurementTrackerToken, hmte);
+    measurementTracker = hmte.product();
+  }
+
+  RectangularEtaPhiTrackingRegion* region = new RectangularEtaPhiTrackingRegion(dirVector, vertexPos,
+                                               minPt, deltaR,
+                                               deltaZ, region_dEta, region_dPhi,
+                                               theOnDemand,
+                                               thePrecise,
+                                               measurementTracker);
+
+  LogDebug("MuonTrackingRegionBuilder")<<"the region parameters are:\n"
+				       <<"\n dirVector: "<<dirVector
+				       <<"\n vertexPos: "<<vertexPos
+				       <<"\n minPt: "<<minPt
+				       <<"\n deltaR:"<<deltaR
+				       <<"\n deltaZ:"<<deltaZ
+				       <<"\n region_dEta:"<<region_dEta
+				       <<"\n region_dPhi:"<<region_dPhi
+				       <<"\n on demand parameter: "<<static_cast<int>(theOnDemand);
+  
+  return region;
+
+}
+
+void MuonTrackingRegionBuilder::fillDescriptions(edm::ParameterSetDescription& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<double>("EtaR_UpperLimit_Par1",0.25);
+  desc.add<double>("DeltaR",0.2);
+  desc.add<edm::InputTag>("beamSpot",edm::InputTag("hltOnlineBeamSpot"));
+  desc.add<int>("OnDemand",-1);
+  desc.add<edm::InputTag>("vertexCollection",edm::InputTag("pixelVertices"));
+  desc.add<double>("Rescale_phi",3.0);
+  desc.add<bool>("Eta_fixed",false);
+  desc.add<double>("Rescale_eta",3.0);
+  desc.add<double>("PhiR_UpperLimit_Par2",0.2);
+  desc.add<double>("Eta_min",0.05);
+  desc.add<bool>("Phi_fixed",false);
+  desc.add<double>("Phi_min",0.05);
+  desc.add<double>("PhiR_UpperLimit_Par1",0.6);
+  desc.add<double>("EtaR_UpperLimit_Par2",0.15);
+  desc.add<edm::InputTag>("MeasurementTrackerName",edm::InputTag("hltESPMeasurementTracker"));
+  desc.add<bool>("UseVertex",false);
+  desc.add<double>("Rescale_Dz",3.0);
+  desc.add<bool>("Pt_fixed",false);
+  desc.add<bool>("Z_fixed",true);
+  desc.add<double>("Pt_min",1.5);
+  desc.add<double>("DeltaZ",15.9);
+  desc.add<double>("DeltaEta",0.2);
+  desc.add<double>("DeltaPhi",0.2);
+  desc.add<int>("maxRegions",1);
+  desc.add<bool>("precise",true);
+  desc.add<edm::InputTag>("input",edm::InputTag("hltL2Muons","UpdatedAtVtx"));

```

Comment:
use ``` InputTag("") ```
visual need for hlt inputs shouldn't show up unnecessarily in all types of configurations